File: RecruitmentSystem/RecruitmentSystem.Domain/Dtos/Internship/InternshipCreateDto.cs
using RecruitmentSystem.Domain.Dtos.Application;
using RecruitmentSystem.Domain.Dtos.Steps;

namespace RecruitmentSystem.Domain.Dtos.Internship;

public class InternshipCreateDto
{
    public string Name { get; set; }
    public string ContactEmail { get; set; }
    public string Address { get; set; }
    public string Description { get; set; }
    public string Requirements { get; set; }
    public bool IsPaid { get; set; }
    public bool IsRemote { get; set; }
    public int SlotsAvailable { get; set; }
    public string Skills { get; set; }
    
    public List<InternshipCreateStepDto> InternshipStepDtos { get; set; }
}

File: frontend/src/interfaces/Internship/InternshipCreateDto.ts
import { InternshipCreateStepDto } from "../Step/InternshipCreateStepDto";
import { SettingCreateDto } from "../Setting/SettingCreate";

export interface InternshipCreateDto {
  name: string;
  contactEmail: string;
  address: string;
  description: string;
  requirements: string;
  isPaid: boolean;
  isRemote: boolean;
  slotsAvailable: number;
  skills: string;
  internshipStepDtos: InternshipCreateStepDto[];
  settingCreateDto: SettingCreateDto;
}
